site/content/entry/yuzu-progress-report-apr-2022/index.md

```diff
@@ -0,0 +1,331 @@
+Additionally, there were some obstacles to implementing it in a direct way, since it requires calling custom code supplied by the game, something which was never needed by any previous service implementation.
+So, {{< gh-hovercard "8164" "some preliminary stubs aside" >}}, byte[] {{< gh-hovercard "8199" "implemented the HLE JIT service" >}} to allow the `Hovercraft` emulator to function and `Super Mario 64` to boot.
+
+{{< gh-hovercard "8261" "In a separate PR" >}}, byte[] adds documentation of how the JIT service interface operates.
+This should help other open source projects, if needed.
+
+Of course, this wasn’t enough to get `Super Mario 64` playable, as there were rendering issues to solve as well.
+
+It’s never that simple… but let's try to explain it simply.
+Nintendo Switch games bundle their own individual GPU driver with each game. 
+This is done to increase compatibility, you don't need to update every console in the world if a driver version has an issue.
+
+For unknown reasons, either the `Hovercraft` emulator or the bundled GPU driver reports Vertex Buffers that are simply too large, especially when compared to what the game actually uses.
+Whether it's an issue in the included emulator or just a driver bug, we can't know for certain, but we do need to work around this problem.
+
+{{< imgs
+	"./vb.png| Erroneous Vertex Buffer size diagram"
+  >}}
+
+So, instead of using the insane reported buffer size, byte[] says NO! and {{< gh-hovercard "8205" "uses the backing memory size" >}} instead.
+
+{{< imgs
+	"./sm64.png| It's-a him! (Super Mario 64)"
+  >}}
+
+Performance on Vulkan is not stellar for now, but you can finally enjoy all 3 of the `Super Mario 3D All-Stars` games with both APIs.
+
+Lastly, [Morph](https://github.com/Morph1984) implemented a fix to {{< gh-hovercard "8135" "keep the web applet open in the foreground" >}}, as the `Super Mario 3D All-Stars` games require it or else they would crash a few minutes into gameplay.
+
+## General graphical fixes
+
+Following up on last month's NVFlinger rewrite, [bunnei](https://github.com/bunnei) continued to track issues and bug reports.
+He fixed the reported issues and further cleaned up the code to improve code quality.
+{{< gh-hovercard "8137" "See the code changes for the NVFlinger rewrite here" >}}.
+
+`Xenoblade Chronicles 2` and `Hyrule Warriors: Age of Calamity` would experience interesting issues which were caused by the new `GPU Garbage Collector` introduced as part of `Project Y.F.C.`. 
+We talked about those changes back in [January](https://yuzu-emu.org/entry/yuzu-progress-report-jan-2022/#other-graphical-fixes).
+
+As you can see below, `Xenoblade Chronicles 2` would use exorbitant amounts of VRAM in OpenGL (Vulkan was unaffected):
+
+{{< single-title-imgs
+	"Not a great way to test your whole VRAM (Xenoblade Chronicles 2)"
```

Comment:
No benefit from having two separate images. Can we combine them into one (before on top, after on bottom) and label them so users are aware?